Add Charmhub documentation

dariofaccin opened the pull request:
# Description

Please include a summary of the change. Please also include relevant motivation and context. List any dependencies that are required for this change.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

gruyaume reviewed (changes requested):
(same comment as in the self-signed-certificates PR)

Good job, this is very thorough.

However I think there is too much text. This is a fairly simple charm at the end of the day and we probably don't want 5-10 pages of text for it.

All the documentation that we add has to be maintained over time and the more concise we are, the less effort we'll have to place down the line.

We should also leverage all the automation we can. For instance, config and integrations are automatically generated in Charmhub. We shouldn't need to write our own docs about these subjects in addition to those.

In general I believe all of our documentation for this charm could hold on 1 page.

dariofaccin commented:
@gruyaume I moved part of the tutorial (deploy, configure, integrate) into a single page. If you want ever less pages, I can moved them right in index page.
I kept the setup and cleanup pages, but these can be removed if you think so.
I removed the relation under "explanation" section since they are already linked in the index.
I would keep the "how to" page since the actions to manually provide a certificate need a precise execution order.
References (actions, configs, integrations, libraries, resources) are linked to the page automatically generated.

saltiyazan commented:
Integration test issue addressed in the [this PR.](https://github.com/canonical/tls-certificates-operator/pull/44)
Take a look, you might want to rebase or wait for that one to be merged :)

gruyaume commented on docs/how-to/h-configure-per-unit.md:
```diff
@@ -0,0 +1,40 @@
+# How to configure per-unit TLS certificates
```
```suggestion
# How to provide TLS certificates to requesting charms
```

dariofaccin replied:
Done.

gruyaume commented on docs/how-to/h-configure-per-unit.md:
```diff
@@ -0,0 +1,40 @@
+# How to configure per-unit TLS certificates
+TLS Certificates Operator allows to configure different certificates for different units of the same desired application.
```
```suggestion
TLS Certificates Operator allows the provisioning of certificates to requiring charms.
```

dariofaccin replied:
Done.

gruyaume commented on docs/index.md:
```diff
@@ -0,0 +1,45 @@
+The TLS Certificates Operator charm is responsible for distributing certificates through relationships. Certificates are provided by the operator through Juju configs.
```
```suggestion
The TLS Certificates Operator charm is responsible for distributing x.509 certificates in contexts where certificates are obtained through a manual process.
```

dariofaccin replied:
Done.

gruyaume commented on docs/index.md:
```diff
@@ -0,0 +1,45 @@
+The TLS Certificates Operator charm is responsible for distributing certificates through relationships. Certificates are provided by the operator through Juju configs.
+
+## Usage
+To deploy the TLS Certificates Operator, all you need to do is run the following command, which will fetch the charm from [Charmhub](https://charmhub.io/tls-certificates-operator) and deploy it to your model:
```
```suggestion
Deploy the charm:
```

dariofaccin replied:
Done.

gruyaume commented on docs/index.md:
````diff
@@ -0,0 +1,45 @@
+
+## Usage
+To deploy the TLS Certificates Operator, all you need to do is run the following command, which will fetch the charm from [Charmhub](https://charmhub.io/tls-certificates-operator) and deploy it to your model:
+```shell
+juju deploy tls-certificates-operator
+```
````
```suggestion
```shell
juju deploy tls-certificates-operator
juju status --watch 1s
```
```

dariofaccin replied:
Done.

gruyaume commented on docs/index.md:
````diff
@@ -0,0 +1,45 @@
+juju deploy tls-certificates-operator
+```
+
+Juju will now fetch TLS Certificates Operator and begin deploying it to the local MicroK8s. This process can take several minutes depending on how provisioned (RAM, CPU, etc) your machine is. You can track the progress by running:
+```shell
+juju status --watch 1s
+```
+The deployment of the charm is completed when its status is in `Active/Idle` state.
````
```suggestion
```

dariofaccin replied:
Done.

gruyaume commented on docs/index.md:
````diff
@@ -0,0 +1,45 @@
+```
+The deployment of the charm is completed when its status is in `Active/Idle` state.
+
+At this point you can relate the TLS Certificates Operator to your charm through the `tls-certificates` interface:
````
Relate to a requirer of TLS certificates:

dariofaccin replied:
Done.

gruyaume commented on docs/index.md:
````diff
@@ -0,0 +1,45 @@
+```shell
+juju relate tls-certificates-operator <your-charm>
+```
+Wait for `juju status --watch 1s` to show all applications/units as `active`.
````
```suggestion
```

dariofaccin replied:
Done.

gruyaume commented on docs/index.md:
```diff
@@ -0,0 +1,45 @@
+| Level | Path                 | Navlink                                                                                                                                                |
+|-------|----------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 1     | how-to               | [How-to guides]()                                                                                                                                      |
+| 2     | h-configure-per-unit | [Configure certificates per unit](/t/tls-certificates-operator-how-to-configure-per-unit-certificates/11612)                                           |
```
I don't especially like the `per unit` wording that we have all over the place. Users use this charm whenever they obtain their certificates in a manual process. The per unit aspect is mostly an implementation detail, not what people care about.

dariofaccin replied:
I removed the `per unit` stuff and replaced with "Provisiong of certificates"
